subsys/partition_manager/multithreading.conf

```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_MULTITHREADING=y
```

Comment:
I know tis is a bit weird but I just couldn't get past https://github.com/nrfconnect/sdk-mcuboot/blob/800d4756702f07364f32538a6f0263c16be3eafc/boot/zephyr/Kconfig#L682 overriding my symbol without doing this.

Another option is that I open another PR to the MCUBoot repository with a change to the Kconfig && !SPI_NOR for the MULTITHREADING.

Reply:
Now there is some time since I've looked at this PR but if I recall correctly there was something in the spi_nor driver which required multithreading to be enabled. I think there was some asynchronous transfer functionality. I'll look into it a bit more properly and get back to you.

Reply:
but you don't seem to have SPI_NOR enabled in mcuboot child image.

At least not in the test I did here: https://github.com/nrfconnect/sdk-nrf/pull/6407#discussion_r795834962

The parent image, SPI_NOR is enabled:
```
$ grep -e 'CONFIG_MULTITHREADING' -e 'CONFIG_SPI_NOR' build/zephyr/.config
CONFIG_SPI_NOR=y
CONFIG_SPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
CONFIG_MULTITHREADING=y
....
```

The MCUboot child image:
```
$ grep -e 'CONFIG_MULTITHREADING' -e 'CONFIG_SPI_NOR' build/mcuboot/zephyr/.config
# CONFIG_MULTITHREADING is not set
# CONFIG_SPI_NOR is not set
CONFIG_SPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
```

so it seems you try to set multithreading in the child because SPI_NOR is enabled in the parent, but you don't ensure that SPI_NOR is enabled in the child image.